src/acceleration/config/AccelerationConfiguration.cpp

```diff
@@ -53,17 +53,22 @@ AccelerationConfiguration::AccelerationConfiguration(
       ATTR_RSLS_REUSED_TIME_WINDOWS("reused-time-windows-at-restart"),
       ATTR_RSSVD_TRUNCATIONEPS("truncation-threshold"),
       ATTR_PRECOND_NONCONST_TIME_WINDOWS("freeze-after"),
+      ATTR_PRECOND_UPDATE("update"),
       VALUE_CONSTANT("constant"),
       VALUE_AITKEN("aitken"),
       VALUE_IQNILS("IQN-ILS"),
       VALUE_IQNIMVJ("IQN-IMVJ"),
       VALUE_QR1FILTER("QR1"),
       VALUE_QR1_ABSFILTER("QR1-absolute"),
       VALUE_QR2FILTER("QR2"),
+      VALUE_QR3FILTER("QR3"),
       VALUE_CONSTANT_PRECONDITIONER("constant"),
       VALUE_VALUE_PRECONDITIONER("value"),
       VALUE_RESIDUAL_PRECONDITIONER("residual"),
       VALUE_RESIDUAL_SUM_PRECONDITIONER("residual-sum"),
+      VALUE_RESIDUAL_SUM_ALWAYS("always"),
+      VALUE_RESIDUAL_SUM_THRESHOLD("on-threshold"),
+      VALUE_RESIDUAL_SUM_LIMIT("limit"),
```

Comment:
If we keep this limit configurable, we need to make both names consistent.
Either "on-limit" and "limit". Or "on-threshold" and "threshold".